### ISSUE TITLE
Fixed an issue that was under certain conditions causing a JVM crash 

### ISSUE THREAD

**Issue by RobAustin** [OpenHFT/Chronicle-Bytes#100]
```

Stack: [0x00007f369b9c0000,0x00007f369bac0000],  sp=0x00007f369babd200,  free space=1012k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
J 1060 C2 net.openhft.chronicle.bytes.BytesInternal.parseUtf8_SB1(Lnet/openhft/chronicle/bytes/NativeBytesStore;JLjava/lang/StringBuilder;I)V (236 bytes) @ 0x00007f367eea553d [0x00007f367eea5440+0xfd]
j  net.openhft.chronicle.bytes.BytesInternal.parseUtf8(Lnet/openhft/chronicle/bytes/RandomDataInput;JLjava/lang/Appendable;I)V+82
j  net.openhft.chronicle.bytes.AppendableUtil.parseUtf8(Lnet/openhft/chronicle/bytes/BytesStore;Ljava/lang/StringBuilder;I)V+17
j  net.openhft.chronicle.bytes.util.UTF8StringInterner.getValue(Lnet/openhft/chronicle/bytes/BytesStore;I)Ljava/lang/String;+18
j  net.openhft.chronicle.bytes.util.UTF8StringInterner.getValue(Lnet/openhft/chronicle/bytes/BytesStore;I)Ljava/lang/Object;+3
j  net.openhft.chronicle.bytes.util.AbstractInterner.intern(Lnet/openhft/chronicle/bytes/BytesStore;I)Ljava/lang/Object;+195
J 1177 C2 net.openhft.chronicle.wire.BinaryWire$BinaryValueIn.text()Ljava/lang/String; (337 bytes) @ 0x00007f367ef2ef80 [0x00007f367ef2ed80+0x200]
J 1222 C2 net.openhft.chronicle.wire.WireMarshaller$StringFieldAccess.setValue(Ljava/lang/Object;Lnet/openhft/chronicle/wire/ValueIn;Z)V (18 bytes) @ 0x00007f367ef0b6c8 [0x00007f367ef0b660+0x68]
J 1315 C2 net.openhft.chronicle.wire.Marshallable.readMarshallable(Lnet/openhft/chronicle/wire/WireIn;)V (7 bytes) @ 0x00007f367ef62198 [0x00007f367ef61fc0+0x1d8]
J 1235 C2 net.openhft.chronicle.wire.Wires.object0(Lnet/openhft/chronicle/wire/ValueIn;Ljava/lang/Object;Ljava/lang/Class;)Ljava/lang/Object; (335 bytes) @ 0x00007f367ef52fb4 [0x00007f367ef521e0+0xdd4]
J 1294 C2 net.openhft.chronicle.wire.VanillaMethodReader.readOne()Z (234 bytes) @ 0x00007f367ef850b0 [0x00007f367ef84700+0x9b0]
j  com.exoduspoint.exd360.common.transport.chronicle.MethodReaderTest.test()V+91
v  ~StubRoutines::call_stub
V  [libjvm.so+0x695b96]  JavaCalls::call_helper(JavaValue*, methodHandle*, JavaCallArguments*, Thread*)+0x1056
V  [libjvm.so+0x9a4677]  Reflection::invoke(instanceKlassHandle, methodHandle, Handle, bool, objArrayHandle, BasicType, objArrayHandle, bool, Thread*)+0x5d7
V  [libjvm.so+0x9a7d87]  Reflection::invoke_method(oopDesc*, Handle, objArrayHandle, Thread*)+0x147
V  [libjvm.so+0x727d4b]  JVM_InvokeMethod+0x26b
j  sun.reflect.NativeMethodAccessorImpl.invoke0(Ljava/lang/reflect/Method;Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;+0
j  sun.reflect.NativeMethodAccessorImpl.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;+100
j  sun.reflect.DelegatingMethodAccessorImpl.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;+6
j  java.lang.reflect.Method.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;+56
j  org.junit.runners.model.FrameworkMethod$1.runReflectiveCall()Ljava/lang/Object;+15
j  org.junit.internal.runners.model.ReflectiveCallable.run()Ljava/lang/Object;+1
j  org.junit.runners.model.FrameworkMethod.invokeExplosively(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;+10
j  org.junit.internal.runners.statements.InvokeMethod.evaluate()V+12
j  org.junit.runners.ParentRunner.runLeaf(Lorg/junit/runners/model/Statement;Lorg/junit/runner/Description;Lorg/junit/runner/notification/RunNotifier;)V+17
j  org.junit.runners.BlockJUnit4ClassRunner.runChild(Lorg/junit/runners/model/FrameworkMethod;Lorg/junit/runner/notification/RunNotifier;)V+30
j  org.junit.runners.BlockJUnit4ClassRunner.runChild(Ljava/lang/Object;Lorg/junit/runner/notification/RunNotifier;)V+6
j  org.junit.runners.ParentRunner$3.run()V+12
j  org.junit.runners.ParentRunner$1.schedule(Ljava/lang/Runnable;)V+1
j  org.junit.runners.ParentRunner.runChildren(Lorg/junit/runner/notification/RunNotifier;)V+44
j  org.junit.runners.ParentRunner.access$000(Lorg/junit/runners/ParentRunner;Lorg/junit/runner/notification/RunNotifier;)V+2
j  org.junit.runners.ParentRunner$2.evaluate()V+8
j  org.junit.runners.ParentRunner.run(Lorg/junit/runner/notification/RunNotifier;)V+20
j  org.junit.runner.JUnitCore.run(Lorg/junit/runner/Runner;)Lorg/junit/runner/Result;+37
j  com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs([Ljava/lang/String;Ljava/lang/String;IZ)I+146
j  com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(Lcom/intellij/rt/execution/junit/IdeaTestRunner;[Ljava/lang/String;Ljava/util/ArrayList;Ljava/lang/String;IZ)I+22
j  com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart([Ljava/lang/String;Ljava/lang/String;Ljava/util/ArrayList;Ljava/lang/String;)I+116
j  com.intellij.rt.execution.junit.JUnitStarter.main([Ljava/lang/String;)V+116
v  ~StubRoutines::call_stub
```

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.17.24](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.17.24), [BOM-2.17.235](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.235)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.17.24](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.17.24), [BOM-2.17.241](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.241)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.101](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
